pkg/controllers/vdb/restart_reconciler.go

```diff
@@ -128,6 +128,16 @@ func (r *RestartReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctr
 		r.Vdb.Spec.InitPolicy != vapi.CommunalInitPolicyScheduleOnly {
 		return r.reconcileCluster(ctx)
 	}
+	// If at least one non-readonly is up when cluster does not have quorum,
+	// we requeue to give that node(s) time to go down before restarting the entire
+	// cluster
+	if !r.PFacts.doesDBHaveQuorum() {
```

Comment:
One issue with this is that it doesn't account for pods that exist but aren't yet part of the database. Consider you start with a 3 node system, then grow the primaries to like 9 nodes. But before we do the add node, we get into here. We may think 6 of the 9 nodes are down and wrongly assume we have lost cluster quorum. I think we need to push this check much lower. There is a spot where we fetch cluster state (fetchClusterNodeState) to prune out nodes that aren't down yet in spread. Perhaps that check needs to expand to allow us to check for cluster quorum?

Reply:
Yes I agree.

Reply:
I added 2 new commits. The older one does what you suggested. On the newer  one I only count a primary if dbExists  to avoid the above situation . Wouldn't that be enough to account for pods that aren't yet part of the database?

Otherwise you can review the older commit and I will keep that as a solution.